src/Service/NamespaceReader.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PharIo\Mediator\Service;
+
+use Composer\IO\IOInterface;
+use function implode;
+use function sprintf;
+use function strpos;
+
+final class NamespaceReader
+{
+	private $io;
+
+	public function __construct(IOInterface $io)
+	{
+		$this->io = $io;
+	}
+
+	public function getNamespace(): string
+	{
+		return $this->readNamespace();
+	}
+
+	private function readNamespace(): string
+	{
+		$query = [
+			'',
+			sprintf(
+				"  <question>%s</question>",
+				'What is the PHP-Namespace for your plugin'
+			),
+			'  <comment>This is in the form "namespace\subnamespace"</comment>',
+			'  # ',
+		];
+
+		while (true) {
+			$answer = $this->io->ask(implode("\n", $query));
+
+			if (false !== strpos($answer, '\\')) {
```

Comment:
- No yoda :-)
- Are we sure that's the only validation we need to determine it's a valid namespace?
- I'm also not sure I understand this validation (maybe it's too late ;) )

Reply:
Yoda removed and validation adapted.